File: Leetcode/python/sol3264.py


#https://leetcode.com/problems/final-array-state-after-k-multiplication-operations-i/
class Solution:

    def priority(self, arr:list)->tuple:
        order = []
        for i, j in enumerate(arr):
            order.append((i,j))
        order.sort(key=lambda x:x[1])
        return order

    def getFinalState(self, nums: List[int], k: int, multiplier: int) -> List[int]:
        for _ in range(k):
            tuples = self.priority(nums)
            index, value = tuples[0]
            nums[index] = multiplier*value


        return nums

File: Leetcode/cpp/sol3264.cpp

//https://leetcode.com/problems/final-array-state-after-k-multiplication-operations-i/description/
class Solution {
public:

    int find_index_value(vector<int>& nums){

        int minimo = numeric_limits<int>::max();
        int index = -1;
        for(auto i = 0; i<nums.size(); i++){
            if(minimo > nums[i]){
                minimo = nums[i];
                index = i;
            }
        }

        return index;
    }
    vector<int> getFinalState(vector<int>& nums, int k, int multiplier) {

        for(auto i = 0; i<k;i++){
            int min = find_index_value(nums);
            nums[min]=nums[min]*multiplier;
        }
        return nums;
    }
};